### DISCUSSION TITLE
Session Provider for Pocketbase

### DISCUSSION THREAD

**Discussion by Nichtmetall** [pocketbase/pocketbase#1215] [Q&A] [answered]
**Hello,** 
I'm very new in Pocketbase and I currently work on a side project.
But I have an Issue:

I am currently logging a user in with this function:
```typescript
export async function login(username: string, password: string) {
    const authData = await database.collection('users').authWithPassword(
        username,
        password,
    );
    router.push("/dashboard");
}
```

and everything is working fine. I can get Data via 
```typescript
export async function getUser(id: string) {
    return await database.collection('users').getOne(id, {
        expand: 'relField1,relField2.subRelField',
    });
}
```

But as soon as I reload the page, I can't access the data again...like the user is logged out.

**Now to my Question:** Is there a way to let the user logged in even if you are reloading and doing stuff?
Only when you are doing nothing for 10mins or click a button you are getting logged out.

**Comment by ganigeorgiev** [comment, marked as answer]
Where the above code is executed - in the browser or in a node server-side handler?

The SDK `AuthStore` uses `LocalStorage` by default and it is available only in the browser.

If your app is using SSR, please check https://github.com/pocketbase/js-sdk#ssr-integration (and even with the SSR there are 2 options - entirely server-side, as shown in the ssr examples, or mixed server and browser).

You can also create your own custom `AuthStore` by extending `BaseAuthStore`. You can find more info in https://github.com/pocketbase/js-sdk#authstore.

**Reply by Nichtmetall** [reply]
Thank you very much!
